src/components/atoms/IconInput/IconInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ChangeEvent } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import "./IconInput.scss";
+
+export interface IconInputProps {
+  className?: string;
+  icon?: IconProp;
+  onChange?: (e?: ChangeEvent<HTMLInputElement>) => void;
+  value?: string;
+  autoComplete?: string;
+  placeholder?: string;
+  iconPosition?: "right" | "left";
+}
+
+const ICON_INPUT_CLASS_NAME = "icon-input";
+const ICON_CLASS_NAME = `${ICON_INPUT_CLASS_NAME}__icon`;
+const INPUT_CLASS_NAME = `${ICON_INPUT_CLASS_NAME}__input`;
+
+export const IconInput: React.FC<IconInputProps> = ({
```

Comment:
I think it might be better to name this something like `Input` or `InputField` or similar rather than `IconInput`. The fact that it can support icons doesn't feel like it needs to be in the component name itself, just having props that support that should be sufficient.

Reply:
renamed: IconInput -> InputField

commit: https://github.com/sparkletown/sparkle/pull/1306/commits/c0acf340a2ba41cb9b7fa7425726ace7838eccf2